new-cep.md

```diff
@@ -0,0 +1,218 @@
+<table>
+  <tr><td> Title </td><td> Conda Generic Plugin Hooks</td>
+  <tr><td> Status </td><td> Draft </td></tr>
+  <tr><td> Author(s) </td><td> Full Name &lt;thathaway@anaconda.com&gt;</td></tr>
+  <tr><td> Created </td><td> Dec 22, 2022</td></tr>
+  <tr><td> Updated </td><td> Dec 22, 2022</td></tr>
+  <tr><td> Discussion </td><td> _TBD_ </td></tr>
+  <tr><td> Implementation </td><td> _TBD_ </td></tr>
+</table>
+
+[conda-pre-invoke-location]: https://github.com/conda/conda/blob/48f51e6c1d412270efbbdb1d9ff571087568b6ea/conda/cli/main.py#L69
+[conda-exception-handler]: https://github.com/conda/conda/blob/48f51e6c1d412270efbbdb1d9ff571087568b6ea/conda/exceptions.py#L1120
+
+## Abstract
+
+In order to support a variety of use cases and extensions to conda's default
+behavior, we propose a set of generic plugin hooks in this CEP. Included will 
+be `pre_command` and `post_command` hooks that will allow
+plugin authors to execute their plugin code before or after conda commands
+run, respectively. Additionally, we introduce the `exception_handler` plugin 
+hook that will allow plugin authors to override and extend behavior of the
+existing [`conda.exceptions.ExceptionHandler`][conda-exception-handler] class. 
+For each hook, we outline example use cases and
+show exactly how plugin authors will define these new hooks.
+
+## Specification
+
+This specification calls for the creation of the following three new plugin hooks:
+
+- `pre_command`: runs before the invoked conda command is run
+- `post_command`: runs after the invoked conda command is run
+- `exception_handler`: allows overriding or extending the existing
+  [`conda.exceptions.ExceptionHandler`][conda-exception-handler]
+
+Below, we discuss an example showing the use of the `pre_command` and `post_command` 
+hooks together and an example of the `exception_handler` hook.
+
+### `pre_command` and `post_command`
+
+Plugin authors create the `pre_command` and `post_command` hooks by first defining a `conda.plugins.hookimpl`
+decorated function called either `conda_pre_command` or `conda_post_command` which return either a `CondaPreCommand` 
+or `CondaPostCommand` class, respectively. These classes will both receive the following three properties:
+
+- `name`: unique name which identifies this plugin hook
+- `action`: a callable which contains the code to be run
+- `run_for`: a Python `set` of strings representing the commands this will be run on (e.g. `install` and `create`)
```

Comment:
is it optional or mandatory parameter? 
(also asked similar question if exception handler hook should have it as well)

Reply:
Good question. I think making this argument optional with the default behavior of it running for every command makes the most sense.